eZ/Publish/API/Repository/Tests/SearchServiceLocationTest.php

```diff
@@ -157,6 +183,67 @@ public function testFieldCollectionContainsNoMatch()
         $this->assertEquals(0, $result->totalCount);
     }
 
+    /**
+     * @covers \eZ\Publish\API\Repository\SearchService::findLocations
+     */
+    public function testNonPrintableUtf8Characters(): void
+    {
+        $folder = $this->createFolderWithNonPrintableUtf8Characters();
+        $query = new LocationQuery(
+            [
+                'query' => new Criterion\Field(
+                    'name',
+                    Criterion\Operator::EQ,
+                    utf8_decode("Non\x09Printable\x0EFolder")
+                ),
+            ]
+        );
+
+        $repository = $this->getRepository();
+        $searchService = $repository->getSearchService();
+        $result = $searchService->findLocations($query);
+
+        $this->assertEquals(1, $result->totalCount);
+        $this->assertEquals(
+            $folder->contentInfo->mainLocationId,
+            $result->searchHits[0]->valueObject->id
+        );
+    }
+
+    /**
+     * @covers \eZ\Publish\API\Repository\SearchService::findLocations
+     * @depends \eZ\Publish\API\Repository\Tests\SearchServiceLocationTest::testNonPrintableUtf8Characters
+     */
+    public function testEscapedNonPrintableUtf8Characters(): void
+    {
+        $setupFactory = $this->getSetupFactory();
+
+        if (
+            !$setupFactory instanceof LegacyElasticsearch AND
+            !$setupFactory instanceof LegacySolrSetupFactory
```

Comment:
Hmm, better. I actually expected `if (!$setupFactory instanceof LegacySolrSetupFactory)` in the first place, as ES has been dropped for 3.0, but this is also acceptable. Almost. Please use `&&` instead of `AND` for consistency's sake and [other issues](https://github.com/kalessil/phpinspectionsea/blob/master/docs/control-flow.md#non-optimal-if-conditions).

Reply:
That what I was expecting from your mythical reviews  :trollface: 

> Please use && instead of AND
Ahhh, old love don't rust